### PR TITLE
sdl client multimonitor support

### PR DESCRIPTION
**Pull request by akallabeth**
based on https://github.com/DanAtIntegrateIT/FreeRDP/tree/integrateit

@DanAtIntegrateIT did take your work and did some small changes to that (so that it works not only on windows ;))
any feedback welcome.

NOTE: This is a basic multimonitor support, so don´t expect flawless operation and fancy monitor layouts to work out of the box.
Also no `gfx` drawing support (eliminating the virtual framebuffer and pushing updates from surface buffer directly to window)

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test again please

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by akallabeth** [comment]
@freerdp-bot test please

File: client/SDL/sdl_kbd.cpp [hunk 1]
```diff
@@ -479,6 +470,26 @@ BOOL sdlInput::keyboard_grab(Uint32 windowID, SDL_bool enable)
 #endif
 }
 
+BOOL sdlInput::mouse_focus(Uint32 windowID)
+{
+	static Uint32 lastWindowID = -1;
```
Comment (hardening) [inline client/SDL/sdl_kbd.cpp:475]: we should probably avoid that static variable and have a member for that ?

Comment (akallabeth) [reply client/SDL/sdl_kbd.cpp:475]: good catch

File: client/SDL/sdl_freerdp.cpp [hunk 2]
```diff
@@ -612,29 +609,53 @@ static BOOL sdl_create_windows(SdlContext* sdl)
...
-
-	for (size_t x = 0; x < windowCount; x++)
+	for (UINT32 x = 0; x < windowCount; x++)
 	{
+		auto monitor = static_cast<rdpMonitor*>(freerdp_settings_get_pointer_array_writable(
+		    sdl->context()->settings, FreeRDP_MonitorDefArray, x));
+
+		Uint32 w = monitor->width;
+		Uint32 h = monitor->height;
+		if (!(sdl->context()->settings->UseMultimon || sdl->context()->settings->Fullscreen))
```
Comment (hardening) [inline client/SDL/sdl_freerdp.cpp:622]: we should alias  `sdl->context()->settings` that is used multiple times here

**Review by hardening — APPROVED** [review]
A few remarks but apart from that looks ok for me.

File: client/SDL/sdl_freerdp.cpp (hunk 2 above)
Comment (akallabeth) [reply client/SDL/sdl_freerdp.cpp]: done

**Comment by akallabeth** [comment]
@freerdp-bot test again please

**Comment by akallabeth** [comment]
@freerdp-bot come on, test properly

**Comment by akallabeth** [comment]
@freerdp-bot bad day? test again...

**Comment by DanAtIntegrateIT** [comment]
Erm... Have I messed something up here... It looks like you've got my multimonitor work, although I have it working for me I didn't feel it was ready for you guys to pick up.... 
It's not far off... but... the code is messy. 



Sometimes my e-mails may be sent outside of normal office hours. An immediate response is not expected. 
To support with well-being please respond when it is convenient for you and within your normal working hours. 
Dan Holliday 
Director 
Integrate-IT Solutions Limited 
T: 01527 908123 
E: ***@***.*** 	

Integrate IT Solutions Limited is a Company registered in England with Company Number 08633599. 
Registered Office: Suite 9, Clive House, Prospect Hill, Redditch, Worcestershire B97 4BY. 
VAT Number: 183 8071 91 

Confidentiality: This e-mail and its attachments are intended for the above named only and may be confidential. If they have come to you in error you must take no action based on them, nor must you copy or show them to anyone; please reply to this e-mail and highlight the error. 
Security Warning: Please note that this e-mail has been created in the knowledge that Internet e-mail is not a 100% secure communications medium. You should remember this lack of security when e-mailing us. 
Viruses: Although we have taken steps to ensure that this e-mail and attachments are free from any virus, we advise that in keeping with good computing practice the recipient should take steps to confirm that they are actually virus free.

**Comment by akallabeth** [comment]
@DanAtIntegrateIT do you have any objections? I´ve looked at the changes you pushed to your branch and did some (small) refactoring on top of that. It works for basic multimonitor setups, but you´re right, not something that is polished, but a good start to do refinements from.

**Comment by DanAtIntegrateIT** [comment]
I don't have any objections at all, I'm chuffed that you're using my work. 
Just don't want to create anyone a headache... So happy for you to carry on. 

Thank you. 

Sometimes my e-mails may be sent outside of normal office hours. An immediate response is not expected. 
To support with well-being please respond when it is convenient for you and within your normal working hours. 
Dan Holliday 
Director 
Integrate-IT Solutions Limited 
T: 01527 908123 
E: ***@***.*** 	

Integrate IT Solutions Limited is a Company registered in England with Company Number 08633599. 
Registered Office: Suite 9, Clive House, Prospect Hill, Redditch, Worcestershire B97 4BY. 
VAT Number: 183 8071 91 

Confidentiality: This e-mail and its attachments are intended for the above named only and may be confidential. If they have come to you in error you must take no action based on them, nor must you copy or show them to anyone; please reply to this e-mail and highlight the error. 
Security Warning: Please note that this e-mail has been created in the knowledge that Internet e-mail is not a 100% secure communications medium. You should remember this lack of security when e-mailing us. 
Viruses: Although we have taken steps to ensure that this e-mail and attachments are free from any virus, we advise that in keeping with good computing practice the recipient should take steps to confirm that they are actually virus free.

**Comment by DanAtIntegrateIT** [comment]
Also, I've just committed by latest updates to my branch as well... 



Sometimes my e-mails may be sent outside of normal office hours. An immediate response is not expected. 
To support with well-being please respond when it is convenient for you and within your normal working hours. 
Dan Holliday 
Director 
Integrate-IT Solutions Limited 
T: 01527 908123 
E: ***@***.*** 	

Integrate IT Solutions Limited is a Company registered in England with Company Number 08633599. 
Registered Office: Suite 9, Clive House, Prospect Hill, Redditch, Worcestershire B97 4BY. 
VAT Number: 183 8071 91 

Confidentiality: This e-mail and its attachments are intended for the above named only and may be confidential. If they have come to you in error you must take no action based on them, nor must you copy or show them to anyone; please reply to this e-mail and highlight the error. 
Security Warning: Please note that this e-mail has been created in the knowledge that Internet e-mail is not a 100% secure communications medium. You should remember this lack of security when e-mailing us. 
Viruses: Although we have taken steps to ensure that this e-mail and attachments are free from any virus, we advise that in keeping with good computing practice the recipient should take steps to confirm that they are actually virus free.

**Pull request merged**